### PR TITLE
feat: INFRA-497 add prometheus orchestration

### PR DESCRIPTION
**Pull request by sebastienbonami**

### CODE REVIEW

File: prometheus/generate.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+create_kustomization_file() {
+  cat <<EOF >kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - namespace.yml
+EOF
+}
+
```
Comment (ihannache) [inline prometheus/generate.sh:16]: here I would put something more concise when it's simple functions.
ex :
append_to_kustomization_file() {
  echo "  - $1" >> kustomization.yaml
}

File: prometheus/generate.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+echo "Downloading Kubernetes CRDs..."
+for crd in "${crds_required[@]}"
+do
+  if wget -q "https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v$operator_version/example/prometheus-operator-crd/monitoring.coreos.com_$crd.yaml" -P crds -N
+  then
+    echo "Successful download of '$crd'."
+    append_to_kustomization_file "crds/monitoring.coreos.com_$crd.yaml"
+  else
+    echo "Failed download of '$crd'."
+  fi
```
Comment (ihannache) [inline prometheus/generate.sh:46]: a practical approach 
return 1

File: prometheus/generate.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+echo "Rendering Helm Template..."
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+if helm template prometheus prometheus-community/kube-prometheus-stack -f values.yml --version=$chart_version > prometheus.yml
+then
+  echo "Successful render."
+  append_to_kustomization_file "prometheus.yml"
+else
+  echo "Failed render."
+fi
```
Comment (ihannache) [inline prometheus/generate.sh:32]: exit 1

File: prometheus/generate.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+echo "Downloading Kubernetes CRDs..."
+for crd in "${crds_required[@]}"
+do
+  if wget -q "https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v$operator_version/example/prometheus-operator-crd/monitoring.coreos.com_$crd.yaml" -P crds -N
+  then
+    echo "Successful download of '$crd'."
+    append_to_kustomization_file "crds/monitoring.coreos.com_$crd.yaml"
+  else
+    echo "Failed download of '$crd'."
```
Comment (ihannache) [inline prometheus/generate.sh:44]: exit 1

**Review by ihannache — APPROVED** [review]

**Review by Magnitus- — APPROVED** [review]
LGTM

**Pull request merged**